src/ray/core_worker/core_worker.h

```diff
@@ -338,11 +338,16 @@ class CoreWorker : public rpc::CoreWorkerServiceHandler {
   ///
   /// \param[in] object_id Object ID corresponding to the object.
   /// \param[in] pin_object Whether or not to pin the object at the local raylet.
+  /// \param[in] generator_id For dynamically created objects, this is the ID
+  /// of the object that wraps the dynamically created ObjectRefs in a
+  /// generator. We use this to notify the owner of the dynamically created
+  /// objects.
   /// \param[in] owner_address Address of the owner of the object who will be contacted by
   /// the raylet if the object is pinned. If not provided, defaults to this worker.
   /// \return Status.
   Status SealExisting(const ObjectID &object_id,
                       bool pin_object,
+                      const ObjectID &generator_id = ObjectID::Nil(),
```

Comment:
Sometimes we are using `const std::optional<ObjectID> &generator_id` and sometimes `ObjectID::Nil()`, let's be consistent?

Reply:
How do we decide when to use which style? What's the guideline we should follow in the future?

Reply:
I think either is fine and I don't feel that it's necessary to have a guideline on it. If you have a strong preference, I can change one of them to the other.